api/src/models/ImageFile.ts

```diff
@@ -79,8 +78,22 @@ class ImageFile {
 
     }
 
-    delete() {
-
+    public delete() {
+        let fs = require("fs");
+        let files: Array<string> = [];
+        for (let size in Object.keys(this.sizes)) {
+            files.push(this.derivativePath(size, "jpg"));
```

Comment:
I don't think you need the png/tiff/txt lines here -- we only create jpg derivatives in the sizes. This method just needs to undo the work done by other methods. You do still need to add the logic to remove OCR files and the original TIFF, however.

Reply:
Updated!

Reply:
Thanks. Nearly there, but for completeness, I think it's also worth deleting `this.derivativePath('ocr/pngs', 'png');` and `this.derivativePath('OCR-DIRTY', 'txt');` for consistency with the Ruby (though as we discussed, I think in practice that's never going to be needed).